### PR TITLE
Remove dead code & perform static code analysis

### PR DESCRIPTION
**Pull request by ogizanagi**
Those are only suggestions, if one of them does not please you, I'll revert the related changes.

### CODE REVIEW

File: src/Command/Setup.php [hunk 1]
```diff
@@ -57,7 +57,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $io->setDecorated(true);
         $io->comment(sprintf('Composing your <info>%s</info> environment', (string) $envType));
 
-        $vars = new Vars($io->ask('Vendor name', null, [$this, 'validateVar']), $io->ask('App name', null, [$this, 'validateVar']));
+        $vars = new Vars(
+            $io->ask('Vendor name', null, [$this, 'validateVar']),
```
Comment (ogizanagi) [inline src/Command/Setup.php]: Calling a constructor multi lined is often better for readability, when the number of arguments exceeds a certain amount, or like in this case, where the arguments are not pre-populated in tmp vars.

File: src/Command/Setup.php [hunk 2]
```diff
@@ -81,7 +84,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
     public function validateVar($value)
     {
         if (!preg_match('/^([-A-Z0-9])*$/i', $value)) {
-            throw new RuntimeException(sprintf('This value must contain only alphanumeric characters and hyphens.', $value));
+            throw new \InvalidArgumentException(sprintf(
```
Comment (ogizanagi) [inline src/Command/Setup.php]: Fixed the exception according to the doc block.

But also suggest to use multilined sprintf declaration for exceptions when the exception construct exceeds 120 chars.
Most of the time, it is sufficient to make the php code sniffer happy, and help for readability :)

File: src/Command/Setup.php [hunk 3]
```diff
@@ -81,7 +84,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
     public function validateVar($value)
```
Comment (ogizanagi) [inline src/Command/Setup.php]: IMHO, this method does validate vars, indeed, but is specific to the `app` & `vendor` vars. Of course those are the only vars now, but this method thus isn't a generic `validateVar` one for further development, or am I wrong ? :/

Above all, I think it should be moved into a dedicated class, but I failed to find a good candidate. WDYT ? :)

File: src/Command/Setup.php (hunk 2 above)
Comment (chalasr) [reply src/Command/Setup.php]: In this case I'm more like:

``` php
throw new \Exception(
    sprintf(...)
);
```

But I'm fine with yours 👍

File: src/Command/Setup.php (hunk 3 above)
Comment (chalasr) [reply src/Command/Setup.php]: Yes, this method bothers me. Maybe a static `Vars::validate()` or something else? As they should be very only used for these vars.

File: src/Process/Build.php [hunk 4]
```diff
@@ -53,9 +54,11 @@ public function __construct($cwd, $tasks = [])
     /**
      * Starts and returns the running process.
      *
+     * @param callable|null $callback
+     *
      * @return \IteratorAggregate
```
Comment (chalasr) [inline src/Process/Build.php]: Could you fix this one? It doesn't return anything anymore (also maybe you prefer have a return like the last exit code)

File: src/Command/Setup.php (hunk 2 above)
Comment (ogizanagi) [reply src/Command/Setup.php]: The idea behind this syntax is that `sprintf` is not that much important and can stay inlined with the exception constructor call, whereas listing the exception message along with the arguments is visually more effective in order to apprehend the exception message placeholders.

File: src/Command/Setup.php (hunk 3 above)
Comment (ogizanagi) [reply src/Command/Setup.php]: Fine. Simply moved this method in the `Vars` class as suggested :)

File: src/Process/Build.php (hunk 4 above)
Comment (ogizanagi) [reply src/Process/Build.php]: Fixed (and returned last exit code)

**Review by chalasr — APPROVED** [review]

**Comment by chalasr** [comment]
Thank you @ogizanagi

**Pull request merged**